backend/src/server/routes/v1/project-router.ts

```diff
@@ -675,4 +677,31 @@ export const registerProjectRouter = async (server: FastifyZodProvider) => {
       return slackConfig;
     }
   });
+
+  server.route({
+    method: "GET",
+    url: "/:workspaceId/folders/project-environments",
```

Comment:
This URL is quite confusing. I thought you were pulling all the environments of a folder. 

Folder is a sub resource for an environment and not the other way around. I would prefer, a bit more clear URL name like `environment-folder-tree`